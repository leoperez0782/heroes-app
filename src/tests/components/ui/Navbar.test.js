import { mount } from "enzyme";
import '@testing-library/jest-dom';
import { MemoryRouter,Router } from "react-router-dom";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";


describe('Pruebas en <Navbar />', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    }
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Lolo'
        }
    };
    const wrapper = mount(
        <MemoryRouter>
            <Router history={historyMock}>

                <AuthContext.Provider value={contextValue}>
                    <Navbar />
                </AuthContext.Provider>
            </Router>
        </MemoryRouter>
    );

    afterEach(()=>{
        jest.clearAllMocks();
    })
    test('deberia de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Lolo');
    });

    test('debe de llamar el logout y usar el history', () => {
        wrapper.find('button').simulate('click');

        expect(contextValue.dispatch).toHaveBeenLastCalledWith({
            type: types.logout
        });

        expect( historyMock.replace).toHaveBeenCalledWith('/login');        
    });



})
