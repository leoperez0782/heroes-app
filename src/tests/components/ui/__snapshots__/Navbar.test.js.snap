// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> deberia de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <div
    className="d-flex align-items-center"
  >
    <span
      className="text-info nav-item nav-link"
    >
      Lolo
    </span>
    <button
      className="nav-item nav-link btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</nav>
`;
